src/main/kotlin/com/coder/gateway/sdk/CoderRestClientService.kt

```diff
@@ -251,53 +252,56 @@ class CoderRestClient(
     }
 }
 
-fun coderSocketFactory(settings: CoderSettingsState) : SSLSocketFactory {
-    if (settings.tlsCertPath.isBlank() || settings.tlsKeyPath.isBlank()) {
-        return SSLSocketFactory.getDefault() as SSLSocketFactory
-    }
+fun SSLContextFromPEMs(certPath: String, keyPath: String, caPath: String) : SSLContext {
```

Comment:
I just refactored this out of `coderSocketFactory` so we could reuse the pem parsing logic for the SSL context for the tests.